pkg/glusterfs/controllerserver.go

```diff
@@ -60,16 +62,50 @@ func (cs *ControllerServer) ParseCreateVolRequest(req *csi.CreateVolumeRequest)
 
 	var reqConf ProvisionerConfig
 	var gdReq api.VolCreateReq
-
+	var err error
 	reqConf.gdVolReq = &gdReq
 
+	replicaCount := defaultReplicaCount
+
 	// Get Volume name
 	if req != nil {
 		reqConf.gdVolReq.Name = req.Name
 	}
+
+	for k, v := range req.GetParameters() {
+
+		switch strings.ToLower(k) {
+
+		case "replicas":
+			replicas := v
+			replicaCount, err = parseVolumeParamInt(replicas, minReplicaCount, maxReplicaCount)
+			if err != nil {
+				return nil, fmt.Errorf("invalid value for parameter 'replicas', value '%s' must be an integer between %d and %d", v, minReplicaCount, maxReplicaCount)
```

Comment:
This should propagate the lower level error back instead of re-write it:
`return nil, fmt.Errorf("invalid value for parameter '%s', %s", k, err)`